test/test_rest.rb

```diff
@@ -23,11 +23,13 @@ def test_generate_telesign_headers_with_post
 
     expected_authorization_header = 'TSA FFFFFFFF-EEEE-DDDD-1234-AB1234567890:' +
         '2xVlmbrxLjYrrPun3G3WMNG6Jon4yKcTeOoK9DjXJ/Q='
+    content_type = "application/x-www-form-urlencoded"
```

Comment:
Doesn't look like you have any test coverage for contentType = "application/json".